src/components/diagram.tsx

```diff
@@ -55,6 +56,28 @@ const loadInitialState = () => {
 
 const dqRoot = DQRoot.create(loadInitialState());
 
+const codapConfig = {
+  customInteractiveStateHandler: true,
+  name: "Quantitative Playground",
+  version: "1.0"
+};
+
+codapInterface.init(codapConfig).then(
+    (o) => {
+      if (o.nodes) {
+        applySnapshot(dqRoot, o);
+      }
+    },
+    (msg) => {
+      console.log( "CODAP not found: " + msg);
+    }
+);
+
+  codapInterface.on("get", "interactiveState", "",
+      () => {
+        return {success: true, values: getSnapshot(dqRoot)};
```

Comment:
I think this will not capture the positions of the nodes correctly.
Currently the node positions are not updated in the MST models as a user drags around the node.
If you look at `copyDiagramURL` you can see how I dealt with that.
You can extract out that part of the code so it can be shared here.
However it is relying on the `rfInstance` state variable which won't be available in this global context.

Reply:
True. Latest changes to node positions are not captured. Will fix.